src/main/java/net/mcreator/ui/MCreatorApplication.java

```diff
@@ -311,11 +311,12 @@ public MCreator openWorkspaceInMCreator(File workspaceFile) {
 						}
 					}
 				}
+
 				this.workspaceSelector.addOrUpdateRecentWorkspace(
-						new RecentWorkspaceEntry(mcreator.getWorkspace(), workspaceFile, Launcher.version.getFullString()));
+						new RecentWorkspaceEntry(mcreator.getWorkspace(), workspaceFile,
+								Launcher.version.getFullString()));
+				return mcreator;

```

Comment:
We don't want to return mcreator if already open workspace is found.

I think addOrUpdateRecentWorkspace should be moved to finally instead

Reply:
The `finally` block is executed before the `return` statement even from the `try` block, right? Or what did you mean?

Reply:
Yes, all you put in finally will execute in any case even if returning from the try block. So if you move addOrUpdateRecentWorkspace there it will be updated in all cases.

Return mcreator should remain where it was.

Analytics placement is better in the PR that in was before too.

Reply:
> The finally block is executed before the return statement even from the try block, right?

Technically it is called right after returning afaik

Reply:
>Analytics placement is better in the PR that in was before too.

So keep it?

Reply:
Yes

Reply:
Before it would track even if workspace was already open

Reply:
✔️ Fixed